test/modules/obsagent-client/src/main.rs

```diff
@@ -0,0 +1,204 @@
+        )
+        .arg(
+            Arg::with_name("push-rate")
+                .short("p")
+                .long("push-rate")
+                .help("Rate at which measurements are pushed out of the client (pushes/sec)")
+        )
+        .arg(
+            Arg::with_name("batching-enabled")
+                .short("b")
+                .long("batching-enabled")
+                .help("Enables or disables batch recording of measurements.")
+        )
+        .arg(
+            Arg::with_name("otlp-endpoint")
+                .short("e")
+                .long("otlp-endpoint")
+                .help("Endpoint to which OTLP messages will be sent.")
+        )
+        .arg(
+            Arg::with_name("enable-value-recorder")
+                .short("r")
+                .long("enable-value-recorder")
+                .help("Enables or disables value recorder instrument.")
+        )
+        .arg(
+            Arg::with_name("enable-value-observer")
+                .short("v")
+                .long("enable-value-observer")
+                .help("Enables or disables value observer instrument.")
+        )
+        .get_matches();
+
+    let config = Config {
+        update_rate: if let Ok(v) = std::env::var("OBS_AGENT_CLIENT_UPDATE_RATE") {
+            v.parse()?
+        } else {
+            value_t!(matches.value_of("update-rate"), f64).unwrap_or(1.0)
+        },
+        push_rate: if let Ok(v) = std::env::var("OBS_AGENT_CLIENT_PUSH_RATE") {
+            v.parse()?
+        } else {
+            value_t!(matches.value_of("push-rate"), f64).unwrap_or(0.2)
+        },
+        batching_enabled: if let Ok(v) = std::env::var("OBS_AGENT_CLIENT_BATCHING_ENABLED") {
+            v.parse()?
+        } else {
+            value_t!(matches.value_of("batching-enabled"), bool).unwrap_or(false)
+        },
+        otlp_endpoint: if let Ok(v) = std::env::var("OBS_AGENT_OTLP_ENDPOINT") {
+            v.parse()?
+        } else {
+            matches
+                .value_of("otlp-endpoint")
+                .unwrap_or("http://localhost:4317")
+                .to_string()
+        },
+        enable_value_recorder: if let Ok(v) =
+            std::env::var("OBS_AGENT_CLIENT_ENABLE_VALUE_RECORDER")
+        {
+            v.parse()?
+        } else {
+            value_t!(matches.value_of("enable-value-recorder"), bool).unwrap_or(false)
+        },
+        enable_value_observer: if let Ok(v) =
+            std::env::var("OBS_AGENT_CLIENT_ENABLE_VALUE_OBSERVER")
+        {
+            v.parse()?
+        } else {
+            value_t!(matches.value_of("enable-value-observer"), bool).unwrap_or(false)
+        },
+    };
+    println!("{:?}", config);
+    Ok(config)
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+    let config = init_config()?;
+    let _started = init_meter(1.0 / config.push_rate, config.otlp_endpoint)?;
+    let meter = global::meter("microsoft.com/azureiot-edge");
```

Comment:
nit: string might be better as const

Applies elsewhere too, but I suppose it is fine since the name has no effect.